core/src/raw/oio/write/multipart_upload_write.rs

```diff
@@ -103,44 +107,116 @@ pub struct MultipartUploadPart {
     pub etag: String,
 }
 
+struct UploadFuture(BoxedFuture<Result<MultipartUploadPart>>);
```

Comment:
How about `type UploadFuture = BoxedFuture<Result<MultipartUploadPart>>`? Seems we don't need a new struct?

And how about using `WritePartFuture` as it's naming?